OKTA-1156505 AI agent-to-app configuration (#6348)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/xaa-agent-to-app/main/index.md

1. Click **Next**.
1. Under **User access and authentication**, check **Allow users to access this agent**, then select one of:

   * **Create a new OIDC app linked to this AI agent:** To create a custom OIDC app integration instance to bind to this AI agent.
   * **Select an existing app:** To select an existing app integration instance in your org to bind to this AI agent. Use this option if your agentic app uses SAML for SSO. See [Create SAML app integrations](https://help.okta.com/oie/en-us/content/topics/apps/apps_app_integration_wizard_saml.htm).
        > **Note:** Currently, you can only select an existing OIN SAML app instance. You can't select an existing OIN OIDC app instance.

   The app you select in the **User access** tab acts as the requesting app role for the XAA flow. It allows your users to sign in to the agentic app through Okta. After the user is signed in, the agentic app can access resource apps on behalf of the signed in user.

1. Click **Next**.
1. Under **Owners**, add owners for the AI agent. This is optional. You can click **Skip for now**, however, Okta recommends that you add least two owners. Click **Save** after you've added the owners.

> 1. Select your AI Agent from the list of AI Agents, and click **Credentials**.

Reviewer: 'client registration' instead of 'credentials'

Author: ```suggestion
1. Select your AI Agent from the list of AI Agents, and click **Client registration**.
```

      1. Click **Done**.
   * Copy the identifier that appears in the **Client ID** field and use it in your AI agent app. This is the requesting app's client ID that's used for OAuth 2.0.
   * Click **Activate**.

## Configure the resource app

For each resource app you want to connect to the AI agent, create a custom or OIN app integration instance in Okta with OIDC or SAML SSO configured. If you use an OIN app as the resource app, it must already have XAA enabled. To enable XAA on a custom app, follow [Enable XAA on a custom app integration](#enable-xaa-on-a-custom-app-integration).

> **Note:** To create a custom SSO app integration instance, see [Create OpenID Connect app integrations](https://help.okta.com/oie/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm) or [Create SAML app integrations](https://help.okta.com/oie/en-us/content/topics/apps/apps_app_integration_wizard_saml.htm) in the product documentation. To create an OIN app integration instance, see [Add existing app integrations](https://help.okta.com/oie/en-us/content/topics/apps/apps-add-applications.htm).

### Enable XAA on a custom app integration

> Configure XXA for an existing custom SSO app integration instance in Okta:

Reviewer: ```suggestion
Configure XAA for an existing custom SSO app integration instance in Okta:
```

#### Create a new OIDC app linked to this AI agent

| &nbsp;  | Custom OIDC app | Custom SAML app | OIN OIDC catalog app | OIN SAML catalog app |
| --- | :---: | :---: | :---: | :---: |
| Creates a new app instance, linked to the AI agent |  ✔  |  ✗  |  ✗  |  ✗  |

This option creates a [custom OIDC app integration instance](https://help.okta.com/okta_help.htm?type=oie&id=create-openid-connect-app-integrations) in your org that's linked to the AI agent. This linked app instance functions as the requesting app (client). You can't create a custom SAML app instance or a new app instance from the OIN catalog.

The linked OIDC app instance is initially deactivated. When you activate your AI agent, the linked app instance is also activated. You can access the app details from the **Applications** page.

#### Select an existing app

| &nbsp;  | Custom OIDC app | Custom SAML app | OIN OIDC catalog app | OIN SAML catalog app |
| ---     | :-------------: | :---: | :---: | :---: |

> | Link to an existing app |  ✗  |  ✔  |  ✗  |  ✔  |

Reviewer: existing custom oidc app is allowed IF you are on the create new agent flow/screen and have not completed agent registration yet (new agent + existing app allowed). but after the agent has been registered, going to its user access tab will no longer allow you to link to an existing oidc app (existing agent + existing app not allowed). not sure if this distinction between the agent itself already existing or not is clear rn

Author: Thanks for this explanation! I'll add it in as a note.

Author: @jamieguo-okta , please take a look at the updates regarding this point.

Reviewer: lgtm, ty!

### Assign users to the requesting app

Assign users to access the AI agent by assigning them to the linked requesting app.

1. Select your AI Agent from the list of **Directory** > **AI Agents**.
1. Click the **User access** tab.
1. Under **User access** > **Users and groups assigned to this agent**, click **Application > Assignments**. The **Assignments** tab appears for your linked SSO app.
1. In the **Assignments** tab, select the users or groups who can access the AI agent.
    See [Assign an app integration to a user](https://help.okta.com/okta_help.htm?type=oie&id=ext-lcm-assign-app-user) and [Assign an app integration to a group](https://help.okta.com/okta_help.htm?type=oie&id=ext-lcm-assign-app-groups) in the product documentation.
    > **Note:** Your linked SSO app is initially inactive, so if you're navigating from the **Applications** page, it may be listed in the **Inactive** tab.

### Activate the AI agent

> You can only activate AI agents that have assigned users.

Reviewer: you don't need assigned users to activate

        * **Resource indicator**: Specify the URLs of the protected resources.
        * **AI agent's client ID registered in this app**: The external client ID of the AI agent registered in the resource app.
        * **Scopes**: Specify the scopes the AI agent is allowed to request from the resource app.

1. Click **Add**.

## Migration from Okta for AI Agent delegation link

If your org is subscribed to Okta for AI Agents, the **Delegations** tab has been renamed to **User access** and **Machine access** on the AI agent page. The **User access** tab contains configuration on the users that can access the AI agent, and the **Machine access** tab contains configuration on the AI agent's non-human access.

In Okta for AI Agent orgs, when you delegate an AI agent to an SSO app, it can only act on a user's behalf if the user is signed in to the app. Previously, you can create multiple delegation links for an AI agent under the **Delegations** tab. However, with the new **User access** configuration, you can only allow users to sign in to one SSO app for an AI agent. You can still have multiple non-user delegation links, which now appear in the **Machine access** tab.

> In the **User access** configuration, if you have an OIDC SSO app linked to your AI agent, it's assumed that the client ID and credentials come from the OIDC app. If you have a SAML SSO app, the client ID and credentials are obtained from the AI agent. In both cases, the requesting client ID and credentials appear in the **Client registration** tab in the AI agent page.

Reviewer: something along these lines maybe?
```suggestion
In the **User access** configuration, if you have an OIDC SSO app linked to your AI agent, it's assumed that the OAuth client ID and credentials are shared with the OIDC app. If you have a SAML SSO app, the OAuth client ID and credentials belong to the AI agent alone. In both cases, the requesting client ID and credentials appear in the **Client registration** tab in the AI agent page.
```

If you've previously configured an AI agent with user-access-app delegation links, they now appear as apps in the **User access** tab. On the tab, a "This agent is using an outdated method for user sign-on" warning message appears. Review the following scenarios to determine if you have to recreate your AI agent or user access links:

* If you have a single SAML user-access-app delegation for your AI agent, you don't need to reconfigure the AI agent's user access app. Your XAA flow still works with the current configuration.

> * If you have a single OIDC user-access-app delegation for your AI agent, delete the AI agent and re-register the AI agent with a new or existing custom OIDC app instance. See [Delete and re-register your AI agent](#delete-and-re-register-your-ai-agent).

Reviewer: they can delete the single existing oidc user-access-app, and recreate a bound/direct user auth one without deleting/re-creating the entire agent, same as the next bullet.

overall the desired end state is the agent has exactly zero or one user SSO app assigned. 
- pre-existing SAML links may count towards that ^, any additional ones should be deleted
- pre-existing OIDC links created before ph2 GA are deprecated and should be deleted in all cases. they can re-create up to one under the new binding model.

Reviewer: actually sorry, they DO still need to delete and re-created the entire agent if they want to link it to the existing oidc app without re-creating the app side. but they don't need to delete the agent if they are ok with linking it to a new oidc app

Author: When I go into the Trex org that Lalitha set up, I'm not able to **Edit** the user access app configuration (the option is greyed out). Therefore, the only way I can link a new or existing OIDC app is to delete the AI agent. @jamieguo-okta : can you confirm this behavior on an SSO-SKU only org with SECURE_AI_XAA?

Reviewer: If we're still talking about the user SSO manual migration scenario for O4AA orgs after XAA rolls out:

- O4AA org without XAA ph2: can create multiple ID token delegation links for user sign on under 'delegations' tab
  <img width="1239" height="801" alt="image" src="https://github.com/user-attachments/assets/8ca810b7-1bcd-4832-bcb0-50f60833a082" />
- O4AA org after XAA ph2 enabled: pre-existing ID token delegation links for user sign on are now shown as outdated under 'user access' tab and can be deleted
  <img width="1225" height="811" alt="image" src="https://github.com/user-attachments/assets/cda724c0-2b8b-4690-a58f-514ef1441140" />
- O4AA org after XAA ph2 enabled and admin has manually migrated to the new binding model: delegation links were deleted and new OIDC binding app has been created, and is now unmodifiable
  <img width="857" height="388" alt="image" src="https://github.com/user-attachments/assets/bfe922c0-9acf-4f20-8314-c0bf266b9551" />
  - (If they already see this^ screen with XAA enabled that means they've migrated to the new model already and don't need to do anything)

If we're only talking about net-new user access configurations after XAA ph2 is rolled out, then yes after binding an OIDC app to an agent, you cannot change it and must delete/recreate it.

Author: It's not net-net XAA ph2. This is for the use case where O4AA + SAML XAA  + XAA SSO was added to the org, now O4AA and SAML XAA is removed. Leaving only XAA SSO available.
In this scenario, users can't bind an existing AI agent with an existing OIDC app. They can only create a new oidc app. If they want to bind to an existing oidc app, they will need to recreate the AI agent. 
I'll add all this into this step. 
@jamieguo-okta , let me know if this is ok.

Reviewer: got it, thanks! i guess it's simpler to just tell them to delete/recreate since that will always work and not need to list out all the specifics.

do you know if there is/will be a doc for the customers that ARE on O4AA and will keep O4AA when XAA rolls out? they will see this impact too

Author: AFAIK, there are no plans for the impact of this migration for O4AA customers.
The O4AA Product docs will be updated to the User access and Machine access tabs.

Author: @jamieguo-okta , can you review L205 to L210? Thank you!

Reviewer: lgtm!

The linked OIDC app instance is initially deactivated. When you activate your AI agent, the linked app instance is also activated. You can access the app details from the **Applications** page.

#### Select an existing app

| &nbsp;  | Custom OIDC app | Custom SAML app | OIN OIDC catalog app | OIN SAML catalog app |
| ---     | :-------------: | :---: | :---: | :---: |
| Link to an existing app |  ✔ Only when registering the AI agent<br>✗ Modifying an existing AI agent  |  ✔  |  ✗  |  ✔  |

This option links the AI agent to an existing app instance in your org. Users can sign in to the linked app to access the AI agent.

You can link to an existing SAML app instance. This existing SAML app instance can be created from an [integration in the OIN catalog](https://help.okta.com/okta_help.htm?type=oie&id=csh-apps-add-app) or from the [Classic experience > Custom SAML App Integration Wizard](https://help.okta.com/okta_help.htm?type=oie&id=csh-apps-aiw-saml).

> During AI agent registration, you can select an existing custom OIDC app instance that was created with the [Classic experience > App Integration Wizard](https://help.okta.com/okta_help.htm?type=oie&id=csh-apps-aiw-main). After the AI agent is registered, you can't modify the user access app to a custom OIDC app instance. You can't link an existing OIN OIDC app instance, during registration or modification.

Reviewer 2: ```suggestion
During AI agent registration, you can select an existing custom OIDC app instance that was created with the [Classic experience > App Integration Wizard](https://help.okta.com/okta_help.htm?type=oie&id=csh-apps-aiw-main). After the AI agent is registered, you can't modify the user access app to a custom OIDC app instance. You can't link an existing OIN OIDC app instance during registration or modification.
```

### Activate the AI agent

1. On the AI agent page, select **Actions** > **Activate**.

> **Note:** Activating your AI agent can take a few seconds. Wait until you see the "AI agent activated successfully" message before you continue configuring or using your AI agent.

You can also activate the AI agent indirectly by activating the linked requesting app.

1. Go to **Applications and Resources** > **Applications**.
1. Select the **Inactive** tab from the **STATUS** column, and find the linked requesting app that you want to activate.
1. Select **Activate** from the dropdown menu beside your app integration. The AI agent is automatically activated when you activate the linked app.

To deactivate the AI agent:

1. On the AI agent page, select **Actions** > **Deactivate**.

> **Notes:**
> * Deactivating your AI agent can take a few seconds. Wait until you see the "AI agent deactivated successfully" message before you continue with other configurations.
> * The rquesting app linked to the AI agent is also deactivated.

## Configure the resource app

> The resource app contains the protected resources that your AI agent wants to access on behalf of the user.

Reviewer 2: ```suggestion
The resource app contains the protected resources that your AI agent can access on behalf of the user.
```

        > **Note:** Ensure that the app you're registering as the resource server supports XAA.
    1. Click **Save**.

## Configure the XAA connection

For each XAA-enbled resource app that you want to connect to your AI agent, configure the connection on the AI agent page:

1. In the Admin Console, go to **Directory** > **AI Agents**.
1. Select an AI agent.
1. Select the **Resource connections** tab.
1. Click **Add resource connection**.
1. From the  **Application** > **Application instance** dropdown, select the XAA-enabled resource app that you configured from [Configure the resource app](#configure-the-resource-app).
    Select **Enable** to grant access to the app through XAA.
    1. Specify the following fields:
        * **Resource indicator**: Specify the URLs of the protected resources.
        * **AI agent's client ID registered in this app**: The external client ID of the AI agent registered in the resource app.
        * **Scopes**: Specify the scopes the AI agent is allowed to request from the resource app.

1. Click **Add**.

## Migration from Okta for AI Agent delegation link

> If your org is subscribed to Okta for AI Agents, the **Delegations** tab has been renamed to **User access** and **Machine access** on the AI agent page. The **User access** tab contains configuration on the users that can access the AI agent, and the **Machine access** tab contains configuration on the AI agent's non-human access.

Reviewer 2: ```suggestion
If your org is subscribed to Okta for AI Agents, the **Delegations** tab has been renamed **User access** and **Machine access** on the AI agent page. The **User access** tab contains configuration on the users that can access the AI agent, and the **Machine access** tab contains configuration on the AI agent's non-human access.
```

* If you have a single SAML app as a user-access delegation link for your AI agent, you don't need to reconfigure the AI agent's user access app. Your XAA flow still works with the current configuration.
* If you have a single OIDC app as a user-access delegation link for your AI agent, delete the AI agent and re-register the AI agent with a new or existing custom OIDC app instance. See [Delete and re-register your AI agent](#delete-and-re-register-your-ai-agent).
* If you previously had multiple apps as user-access delegation links, delete the existing user-access apps and recreate only one user-access app for linking. See [Delete your previous user access apps](#delete-previous-user-access-apps), then see [Reconfigure direct user authentication for your AI agent](#reconfigure-direct-user-authentiation-for-your-ai-agent).
* If you want to modify your previous delegation configuration, see [Reconfigure direct user authentication for your AI agent](#reconfigure-direct-user-authentiation-for-your-ai-agent).

### Delete previous user access apps

1. In the Admin Console, go to **Directory** > **AI Agents**.
1. Select an AI agent.
1. Select the **User access** tab.

> 1. In the **User sign-on** section, click the more icon (![three-dot more icon](/img/icons/odyssey/more.svg)) next to the user-access app that you want to remove, then select **Delete**.

Reviewer 2: just checking this format - (!

Author: This should be ok. That's the format we use for adding images.:
<img width="843" height="76" alt="image" src="https://github.com/user-attachments/assets/d293728e-cc1b-4719-9500-95bbde7aed3c" />

If you've previously configured an AI agent with user access delegation links, they now appear as apps in the **User access** tab. On the tab, a "This agent is using an outdated method for user sign-on" warning message appears.

Review the following scenarios to determine if you have to reconfigure your AI agent or user access links:

* If you have a single SAML app as a user-access delegation link for your AI agent, you don't need to reconfigure the AI agent's user access app. Your XAA flow still works with the current configuration.
* If you have a single OIDC app for your AI agent's user-access delegation link, you have two options depending on what configuration you want to preserve.
    * If you want to keep your AI agent configuration and can reconfigure a new OIDC app integration for linking:
        1. Delete your linked app (see [Delete your previous user access apps](#delete-previous-user-access-apps)).
        1. Relink a user access app by selecting the **Create a new OIDC app linked to this agent** option (see [Reconfigure direct user authentication for your AI agent](#reconfigure-direct-user-authentiation-for-your-ai-agent)).

>     * If you want to keep your OIDC app configuration and can reconfigure the AI agent:

Reviewer 2: because this is only one step, could it be a single sentence?

If you want to keep your OIDC app configuration and can reconfigure the AI agent, [Delete and re-register your AI agent](#delete-and-re-register-your-ai-agent) and link to the existing OIDC app.

Author: Hi @grahamsmith-okta , I added your suggestion. Can I please get another ✅ ?
